### PR TITLE
Directions functionality

### PR DESCRIPTION
**Pull request by Jord-JD**
This PR adds functionality to retrieve directions information via the Google Maps API. It supports travel via driving, walking, bicycling, and public transport.

Example usage:

```php
$address1 = "10 Downing St, Westminster, London SW1A UK";
$address2 = "Schott House, Drummond Rd, Stafford ST16 3EL";

$journey = $simpleGoogleMaps->directions($address1, $address2, TravelMode::DRIVING);

foreach($journey as $step) {
    echo $step->duration.' secs  ';
    echo "\t";
    echo $step->distance.' m    ';
    echo "\t";
    echo $step->description;
    echo PHP_EOL;
}

echo 'Totals: '.$journey->duration().' secs, '.$journey->distance().' km';
echo PHP_EOL;
```

This will produce output similar to the following.

```
134 secs        452 m           Head north on Whitehall / A3212 toward Horse Guards Ave May be closed at certain times or days 
203 secs        1029 m          At the roundabout, take the 1st exit onto The Mall Parts of this road may be closed at certain times or days 
121 secs        688 m           Turn right onto Constitution Hill 
34 secs         141 m           Turn left onto Duke of Wellington Pl Leaving toll zone 
20 secs         83 m            Turn right onto Grosvenor Pl 
21 secs         107 m           Slight right onto Piccadilly May be closed at certain times or days 
164 secs        1244 m          Slight left onto Park Ln / A4202 
35 secs         199 m           Slight left onto Cumberland Gate 
16 secs         68 m            Turn right onto Bayswater Rd 
92 secs         410 m           Slight left onto Edgware Rd / A5 Entering toll zone in 280 m at Upper Berkeley St Leaving toll zone in 300 m at Stourcliffe St 
48 secs         177 m           Turn right onto George St Entering toll zone 
152 secs        531 m           Turn left onto Seymour Pl 
46 secs         231 m           Turn left onto Marylebone Rd / A501 Leaving toll zone 
722 secs        9209 m          Keep right to continue on Marylebone Flyover / A40 Continue to follow A40 
708 secs        14581 m         Keep right to continue on Western Ave / A40 
4636 secs       140047 m        Keep right to continue on M40 , follow signs for M25 / Birmingham / Oxford / Beaconsfield 
49 secs         1164 m          At junction 3A , take the M42 / Railway Station / Airport exit to M1 / M6 / Birmingham (E, N & C) / Solihull / N.E.C. 
856 secs        22091 m         Merge onto M42 
14 secs         375 m           Keep right at the fork to continue on M6 Toll 
1121 secs       33363 m         Keep right at the fork to stay on M6 Toll Toll road 
756 secs        19905 m         Continue onto M6 
23 secs         206 m           At junction 14 , take the A34 exit to Stone / Stafford (N) 
103 secs        1213 m          At the roundabout, take the 3rd exit onto A34 Go through 1 roundabout 
80 secs         925 m           At the roundabout, take the 2nd exit onto Beaconside / A513 
128 secs        1550 m          Turn right onto Common Rd 
13 secs         123 m           Turn left onto Astonfields Rd 
76 secs         395 m           Turn left onto Drummond Rd Destination will be on the left 
Totals: 10371 secs, 250507 km
```

### CODE REVIEW

**Comment by jaredkove** [comment]
I think I might find changing the class name of `LatLong` to be `GeographicalPosition` or similar might make sense. My thinking is that you could then use that same class across any other location libraries we may happen to make in the future.

**Review by dextermb — CHANGES_REQUESTED** [review]
Currently the classes are not PSR-2 compliant. Just seems to be issues with new lines. Check out the [example](https://www.php-fig.org/psr/psr-2/#11-example).

Namespaces should go directly under the opening PHP tag:
```php
<?php
namespace Langleyfoxall;

class Example
{
    // ...
}
```

There also seems to be some spacing between the `namespace` and `use` in the likes of the  `Journey` class.

**Comment by Jord-JD** [comment]
@wheatleyjj That's a good idea. I think it's possible one for a future PR though. Maybe create an issue related to the idea of separating that class out into its own package?

**Comment by Jord-JD** [comment]
@dextermb Blank lines before namespace have been removed.

**Review by dextermb — APPROVED** [review]

File: src/Objects/SimpleGoogleMaps.php [hunk 1]
```diff
@@ -1,153 +1,236 @@
...
+     */
+    public function directions($from, $to, $travelMode = TravelMode::DRIVING)
+    {
+        if (is_object($from) && get_class($from) === LatLong::class) {
+            $from = $from->lat.','.$from->long;
+        }
+
+        if (is_object($to) && get_class($to) === LatLong::class) {
+            $to = $to->lat.','.$to->long;
+        }
```
Comment (robmilward) [inline src/Objects/SimpleGoogleMaps.php]: Nothing breaking at all, but is it worth letting the LatLong object determine what it's own string representation looks like? Comma separated LatLong pairs are fairly commonplace and bound to be used elsewhere.

Just out of curiosity, is there any intentional benefit of performing `is_object($from) && get_class($from) === LatLong::class` over `$from instanceof LatLong`?

**Review by robmilward — CHANGES_REQUESTED** [review]

File: src/Objects/SimpleGoogleMaps.php (hunk 1 above)
Comment (Jord-JD) [reply src/Objects/SimpleGoogleMaps.php]: Changed. Good call.

**Review by robmilward — APPROVED** [review]

**Review by NilesB — COMMENTED** [review]
The [Google Directions API](https://developers.google.com/maps/documentation/directions/intro#DirectionsRequests) has optional parameters that a lot of people would probably want such as `waypoints`, `language` and `traffic_mode`.

Perhaps users should be able to pass in an arbitrary array that will be url encoded and added to the request.

**Comment by Jord-JD** [comment]
@NilesB Nice idea. Would you mind converting it to an issue?

**Comment by NilesB** [comment]
@DivineOmega Sure

**Pull request merged**